docs/contracts.rst

```diff
@@ -990,12 +1020,28 @@ virtual method lookup.
 
 .. index:: ! constructor
 
+.. _constructor:
+
 Constructors
 ============
-A constructor is an optional function declared with the ``constructor`` keyword which is executed upon contract creation.
-Constructor functions can be either ``public`` or ``internal``. If there is no constructor, the contract will assume the
-default constructor: ``contructor() public {}``.
 
+A constructor is an optional function declared with the ``constructor`` keyword
+which is executed upon contract creation, and where you can run contract
+initialisation code.
+
+Other code executed once right before the constructor always generates code to
+represent the initial storage state, initialising any state variables you
```

Comment:
`represent`?

Reply:
@chriseth Yeah. Do you think that's too ambiguous? It seems OK to me, but I could find something stronger.

Reply:
I just don't understand how code can represent state. What about `code that initialises state variables if required. Note that a newly created contract has a state of all zeros, which is the default value for all variables.` or something like that.